### PR TITLE
Data Sync Center Support Etcd

### LINKED ISSUE #1545: Data Sync Center Support Etcd
## Task

Data Sync Center Support consul

Task List:

- [ ] Data Sync Center Support consul

Please read the [Contribution Guideline](https://dromara.org/projects/soul/contributor/) before submitting the PR

### PR DESCRIPTION
**Pull request by lw1243925457**
// Describe your PR here; eg. Fixes #1545 

Make sure that:

- [x] You have read the [contribution guidelines](https://dromara.org/projects/soul/contributor/).
- [x] You submit test cases (unit or integration tests) that back your changes.
- [x] Your local test passed `mvn clean install -Dmaven.javadoc.skip=true`.

### CODE REVIEW

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/config/DataSyncConfiguration.java [hunk 1]
```diff
@@ -168,5 +173,47 @@ public ServerEndpointExporter serverEndpointExporter() {
...
+
+        /**
+         * Config event listener data changed listener.
+         *
+         * @param etcdClient the etcd client
+         * @return the data changed listener
+         */
+        @Bean
+        @ConditionalOnMissingBean(EtcdDataDataChangedListener.class)
+        public DataChangedListener zookeeperDataChangedListener(final EtcdClient etcdClient) {
```
Comment (yu199195) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/config/DataSyncConfiguration.java:201]: pls refactor method  name

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/config/DataSyncConfiguration.java [hunk 2]
```diff
@@ -168,5 +173,47 @@ public ServerEndpointExporter serverEndpointExporter() {
...
+        /**
+         * data init.
+         *
+         * @param etcdClient        the etcd client
+         * @param syncDataService the sync data service
+         * @return the etcd data init
+         */
+        @Bean
+        @ConditionalOnMissingBean(EtcdDataInit.class)
+        public EtcdDataInit zookeeperDataInit(final EtcdClient etcdClient, final SyncDataService syncDataService) {
```
Comment (yu199195) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/config/DataSyncConfiguration.java:214]: pls refactor method  name

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/listener/etcd/EtcdDataDataChangedListener.java [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+import org.apache.shenyu.common.enums.DataEventTypeEnum;
+import org.apache.shenyu.common.utils.GsonUtils;
+
+import java.net.URLEncoder;
+import java.util.List;
+
+/**
+ * EtcdDataDataChangedListener.
+ *
+ * @author liuwei
```
Comment (yu199195) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/listener/etcd/EtcdDataDataChangedListener.java:38]: remove author

File: shenyu-common/src/main/java/org/apache/shenyu/common/constant/EtcdPathConstants.java [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.shenyu.common.constant;
+
+/**
+ * Etcd PathConstants.
+ */
+public final class EtcdPathConstants implements Constants {
```
Comment (yu199195) [inline shenyu-common/src/main/java/org/apache/shenyu/common/constant/EtcdPathConstants.java:23]: this alreay exist?

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/config/properties/EtcdProperties.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+package org.apache.shenyu.admin.config.properties;
+
+import lombok.Data;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+
+/**
+ * etcd config.
+ *
+ * @author liuwei
```
Comment (dengliming) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/config/properties/EtcdProperties.java:26]: We should remove `@author`

Comment (lw1243925457) [reply shenyu-admin/src/main/java/org/apache/shenyu/admin/config/properties/EtcdProperties.java:26]: OK，i wiil remove author

File: shenyu-common/src/main/java/org/apache/shenyu/common/constant/EtcdPathConstants.java (hunk 4 above)
Comment (lw1243925457) [reply shenyu-common/src/main/java/org/apache/shenyu/common/constant/EtcdPathConstants.java:23]: ok, how about move ZKPathConstants to CommonPathConstants, let etcd and zookeeper use the path constants？

**Review by yu199195 — APPROVED** [review]

**Pull request merged**